flink-ml-lib/src/main/java/org/apache/flink/ml/feature/univariatefeatureselector/UnivariateFeatureSelector.java

```diff
@@ -0,0 +1,301 @@
+        DataStream<UnivariateFeatureSelectorModelData> modelData =
+                tEnv.toDataStream(output)
+                        .transform(
+                                "selectIndicesFromPValues",
+                                TypeInformation.of(UnivariateFeatureSelectorModelData.class),
+                                new SelectIndicesFromPValuesOperator(
+                                        getSelectionMode(), getActualSelectionThreshold()))
+                        .setParallelism(1);
+        UnivariateFeatureSelectorModel model =
+                new UnivariateFeatureSelectorModel().setModelData(tEnv.fromDataStream(modelData));
+        ReadWriteUtils.updateExistingParams(model, getParamMap());
+        return model;
+    }
+
+    private double getActualSelectionThreshold() {
+        Double threshold = getSelectionThreshold();
+        if (threshold == null) {
+            String selectionMode = getSelectionMode();
+            if (NUM_TOP_FEATURES.equals(selectionMode)) {
+                threshold = 50.0;
+            } else if (PERCENTILE.equals(selectionMode)) {
+                threshold = 0.1;
+            } else {
+                threshold = 0.05;
+            }
+        } else {
+            if (NUM_TOP_FEATURES.equals(getSelectionMode())) {
+                Preconditions.checkArgument(
+                        threshold >= 1 && threshold.intValue() == threshold,
+                        "SelectionThreshold needs to be a positive Integer "
+                                + "for selection mode numTopFeatures, but got %s.",
+                        threshold);
+            } else {
+                Preconditions.checkArgument(
+                        threshold >= 0 && threshold <= 1,
+                        "SelectionThreshold needs to be in the range [0, 1] "
+                                + "for selection mode %s, but got %s.",
+                        getSelectionMode(),
+                        threshold);
+            }
+        }
+        return threshold;
+    }
+
+    private static class SelectIndicesFromPValuesOperator
+            extends AbstractStreamOperator<UnivariateFeatureSelectorModelData>
+            implements OneInputStreamOperator<Row, UnivariateFeatureSelectorModelData>,
+                    BoundedOneInput {
+        private String selectionMode;
+        private double threshold;
+
+        private List<Tuple2<Double, Integer>> pValuesAndIndices;
+        private ListState<List<Tuple2<Double, Integer>>> pValuesAndIndicesState;
+
+        public SelectIndicesFromPValuesOperator(String selectionMode, double threshold) {
+            this.selectionMode = selectionMode;
+            this.threshold = threshold;
+        }
+
+        @Override
+        public void endInput() {
+            List<Integer> indices = new ArrayList<>();
+
+            switch (selectionMode) {
+                case NUM_TOP_FEATURES:
+                    pValuesAndIndices.sort(Comparator.comparing(t -> t.f0));
```

Comment:
Let's add comparisons to indices when pValues are the same. And let's add test cases to verify the selected indices in this situation.